app/models/namespace.rb

```diff
@@ -1,8 +1,15 @@
 class Namespace < ActiveRecord::Base
 
+  NAME_ALLOWED_CHARS = 'a-z0-9\-_'
+
   has_many :repositories
   belongs_to :registry
   belongs_to :team
-  validates :name, presence: true
+  validates :name, presence: true,
+                   format: { with: /\A[#{NAME_ALLOWED_CHARS}]+\Z/, message: 'Only allowed letters: [a-z0-9-_]' }
```

Comment:
Nitpicking: maybe you could write the `message` key on the next line, just like you did in the user.rb file :)